src/load.rs

```diff
@@ -47,14 +40,29 @@ impl<'a> eval::Env for BuildImplicitVars<'a> {
 }
 
 /// Internal state used while loading.
-#[derive(Default)]
 pub struct Loader {
-    pub graph: graph::Graph,
+    _graph: Rc<RefCell<graph::Graph>>,
```

Comment:
I had to wrap the graph like this, so I can share it with `file_loader`. Not sure how kosher that is, but I felt not having to remember about converting `FileId` to `PathBuf` at the site is nice.